docs/usage/training_time_compression/pruning/Usage.md

```diff
@@ -0,0 +1,135 @@
+# Pruning
+
+## Non-Structured Pruning
+
+Sparsity algorithm zeros weights in a non-structured way, so that zero values are randomly distributed inside
+the tensor. Most of the pruning algorithms set the less important weights to zero but the criteria of how they
+do it is different. The framework contains several implementations of pruning methods.
+
+### Magnitude
+
+The magnitude pruning method implements a naive approach that is based on the assumption that the contribution
+of lower weights is lower so that they can be pruned. After each training epoch the method calculates a threshold based
+on the current pruning ratio and uses it to zero weights which are lower than this threshold.
+
+And here there are two options:
+
+* `UNSTRUCTURED_MAGNITUDE_LOCAL`: Unstructured magnitude-based pruning with local importance calculation.
+    Weight importance is computed independently for each tensor.
+
+* `UNSTRUCTURED_MAGNITUDE_GLOBAL`: Unstructured magnitude-based pruning with global importance calculation.
+    Weight importance is computed across all tensors selected for pruning.
+
+```python
+import nncf
+
+...
+
+pruned_model = nncf.prune(
+    model,
+    mode=nncf.PruneMode.UNSTRUCTURED_MAGNITUDE_GLOBAL,
+    ratio=0.7,
+    examples_inputs=example_input,
+)
+```
+
+To get a more accurate model, it is recommended to fine-tune the model for several epochs or use batch norm adaptation.
+
+#### Batch Norm Adaptation after Pruning
+
+When using magnitude pruning without fine-tuning, it is recommended to perform Batch Norm adaptation after pruning to get more accurate results.
+
+```python
+import nncf
+
+...
+
+def transform_fn(batch: tuple[torch.Tensor, int]) -> torch.Tensor:
+    inputs, _ = batch
+    return inputs.to(device=device)
+
+calibration_dataset = nncf.Dataset(train_loader, transform_func=transform_fn)
+
+pruned_model = nncf.batch_norm_adaptation(
+    pruned_model,
+    calibration_dataset=calibration_dataset,
+    num_iterations=200,
+)
+```
+
+## Regularization-Based
+
+The method is based on $L_0$-regularization, with which parameters of the model tend to zero:
+
+$||\theta||\_0 = \sum\limits_{i=0}^{|\theta|} \lbrack \theta\_i = 0 \rbrack$
+
+We then reparametrize the network's weights as follows:
+
+$\theta_{sparse}^{(i)} = \theta_i \cdot \epsilon_i, \quad \epsilon_i \sim \mathcal{B}(p_i)$
+
+Here, $\mathcal{B}(p_i)$ is the Bernoulli distribution, $\epsilon_i$ may be interpreted as a binary mask that selects which weights should be zeroed. We then add the regularizing term to the objective function that encourages desired level of sparsity to our model:
+
+$L_{sparse} = \mathbb{E}\_{\epsilon \sim P_{\epsilon}} \lbrack \frac{\sum\limits_{i=0}^{|\theta|} \epsilon_i}{|\theta|} - level \rbrack ^2 $
```

Comment:
isn't it 1-level?
https://arxiv.org/pdf/2002.08679
<img width="812" height="270" alt="Image" src="https://github.com/user-attachments/assets/9995bd73-a085-49d5-b03f-c76c05438b29" />

Reply:
we have `target_ratio`. isn't it the same as sparsity_level?
if so, need to fix formula in the docs:
https://github.com/openvinotoolkit/nncf/blob/ebed81c055455ad250f12f87080c1bc76e29c5a5/src/nncf/torch/function_hook/pruning/rb/losses.py#L70